ruma-api-macros/src/api.rs

```diff
@@ -227,19 +242,39 @@ pub fn expand_all(api: Api) -> syn::Result<TokenStream> {
     let error = &api.error_ty;
     let request_lifetimes = api.request.combine_lifetimes();
 
-    let non_auth_endpoint_impls = if authentication != "None" {
-        TokenStream::new()
-    } else {
-        quote! {
-            #[automatically_derived]
-            impl #request_lifetimes #ruma_api::OutgoingNonAuthRequest
-                for Request #request_lifetimes
-            {}
+    let mut non_auth_endpoint_impls = vec![];
+    for auth in &api.metadata.authentication {
```

Comment:
I think this can be simplified to

```rust
lte non_auth_endpoint_impls: TokenStream = api.metadata
	.authentication
    .map(|auth| if auth.value != "None" {
        TokenStream::new()
    } else {
        // ...
    })
    .collect();